core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -419,7 +419,7 @@ private[kafka] class Processor(val id: Int,
     "socket-server",
     metricTags,
     false,
-    ChannelBuilders.serverChannelBuilder(securityProtocol, channelConfigs, credentialProvider.credentialCache))
+    ChannelBuilders.serverChannelBuilder(listenerName, securityProtocol, config, credentialProvider.credentialCache))
```

Comment:
I think it's OK to leave it as is as we fallback to the non-prefixed version. This is confirmed by the fact that all the current tests (most of which rely on the default security protocols) pass.